frontend/components/model/js/model.component.ts

```diff
@@ -162,44 +162,55 @@ class ModelController implements ng.IController {
 		this.RevisionsService.listAll(this.account, this.model);
 
 		if (!this.ViewerService.currentModel.model) {
-			console.debug("Initiating Viewer");
 			if (this.ViewerService.viewer) {
-				this.ViewerService.initViewer()
-					.then(() => {
-						this.ViewerService.loadViewerModel(
-							this.account,
-							this.model,
-							this.branch,
-							this.revision,
-						);
-					});
+				this.ViewerService.initViewer().then(() => {
+					this.loadModel();
+				}).catch((err) => {
+					console.error("Failed to load model: ", err);
+				});
+			} else {
+				console.error("Failed to locate viewer");
 			}
 		} else {
-			// Load the model
-			this.ViewerService.loadViewerModel(
-				this.account,
-				this.model,
-				this.branch,
-				this.revision,
-			);
+			this.loadModel();
 		}
+	}
+
+	public setSelectedObjects(selectedObjects) {
+		this.selectedObjects = selectedObjects;
+	}
+
+	public setInitialSelectedObjects(data) {
+		this.initialSelectedObjects = data.selectedObjects;
+		// Set the value to null so that it will be registered again
+		this.$timeout(() => {
+			this.initialSelectedObjects = null;
+		});
+	}
+
+	private loadModel() {
+		this.ViewerService.loadViewerModel(
+			this.account,
+			this.model,
+			this.branch,
+			this.revision,
+		).then( () => {
+			// IMPORTANT: only load model settings after it has started loading the model
+			// loadViewerModel can cancel previous model loads which will kill off old unity promises
+			this.loadModelSettings();
+		});
+	}
 
+	private loadModelSettings() {
 		this.ViewerService.getModelInfo(this.account, this.model)
 			.then((response) => {
-				const data = response.data;
-				this.settings = data;
-
-				const isFederation = data.federate;
-				if (isFederation) {
-					this.PanelService.hideSubModels(this.issuesCardIndex, false);
-				} else {
-					this.PanelService.hideSubModels(this.issuesCardIndex, true);
-				}
+				this.settings = response.data;
+				this.PanelService.hideSubModels(this.issuesCardIndex, !this.settings.federate);
 
-				this.ViewerService.updateViewerSettings(data.updateViewerSettings);
-				this.ClipService.initClip(data.properties.unit);
+				this.ViewerService.updateViewerSettings(this.settings);
```

Comment:
Shouldn't it be `this.settings.updateViewerSettings`?

Reply:
no, that's exactly why we weren't passing down the units to unity before (thus maps were broken)